From f38d1f74af7a5ef0ccee2029be0d8036b9020f1a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Tue, 30 Oct 2018 18:26:10 +0200
Subject: fix xattr hash to be sha1

The hash type was accidentally changed in previous commit. Currently
csum->data cannot hold longer hash, so fix the hash.
---
 src/io.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/io.c b/src/io.c
index 0295807..382fd1b 100644
--- a/src/io.c
+++ b/src/io.c
@@ -632,7 +632,7 @@ void apk_fileinfo_hash_xattr_array(struct apk_xattr_array *xattrs, const EVP_MD
 
 	qsort(xattrs->item, xattrs->num, sizeof(xattrs->item[0]), cmp_xattr);
 
-	EVP_DigestInit_ex(mdctx, EVP_sha256(), NULL);
+	EVP_DigestInit_ex(mdctx, EVP_sha1(), NULL);
 	foreach_array_item(xattr, xattrs) {
 		hash_len_data(mdctx, strlen(xattr->name), xattr->name);
 		hash_len_data(mdctx, xattr->value.len, xattr->value.ptr);
-- 
cgit v1.1

